bazel/cel-cpp.patch

```diff
@@ -1,168 +1,61 @@
-index 80ac45c..7dceb93 100644
---- a/eval/public/portable_cel_expr_builder_factory.cc
-+++ b/eval/public/portable_cel_expr_builder_factory.cc
-@@ -20,6 +20,7 @@
- #include <string>
- #include <utility>
- 
-+#include "absl/log/log.h"
- #include "absl/status/status.h"
- #include "eval/compiler/flat_expr_builder.h"
- #include "eval/public/cel_options.h"
-diff --git a/eval/public/structs/BUILD b/eval/public/structs/BUILD
-index 9187518..5151bb0 100644
---- a/eval/public/structs/BUILD
-+++ b/eval/public/structs/BUILD
-@@ -192,7 +192,6 @@ cc_library(
-     hdrs = ["legacy_type_provider.h"],
-     deps = [
-         ":legacy_type_adapter",
--        "//base:type_provider",
-         "@com_google_absl//absl/types:optional",
-     ],
- )
-diff --git a/eval/public/structs/field_access_impl.cc b/eval/public/structs/field_access_impl.cc
-index 788a476..e4b70b3 100644
---- a/eval/public/structs/field_access_impl.cc
-+++ b/eval/public/structs/field_access_impl.cc
-@@ -25,6 +25,7 @@
- #include "google/protobuf/arena.h"
- #include "google/protobuf/map_field.h"
- #include "absl/container/flat_hash_set.h"
-+#include "absl/log/log.h"
- #include "absl/status/status.h"
- #include "absl/status/statusor.h"
- #include "absl/strings/str_cat.h"
-diff --git a/eval/public/structs/legacy_type_provider.h b/eval/public/structs/legacy_type_provider.h
-index b1623fc..d3d88d6 100644
---- a/eval/public/structs/legacy_type_provider.h
-+++ b/eval/public/structs/legacy_type_provider.h
-@@ -16,7 +16,6 @@
- #define THIRD_PARTY_CEL_CPP_EVAL_PUBLIC_STRUCTS_TYPE_PROVIDER_H_
- 
- #include "absl/types/optional.h"
--#include "base/type_provider.h"
- #include "eval/public/structs/legacy_type_adapter.h"
- 
- namespace google::api::expr::runtime {
-@@ -25,8 +24,10 @@ namespace google::api::expr::runtime {
- //
- // Note: This API is not finalized. Consult the CEL team before introducing new
- // implementations.
--class LegacyTypeProvider : public cel::TypeProvider {
-+class LegacyTypeProvider {
-  public:
-+  virtual ~LegacyTypeProvider() = default;
-+
-   // Return LegacyTypeAdapter for the fully qualified type name if available.
-   //
-   // nullopt values are interpreted as not present.
-@@ -45,7 +46,7 @@ class LegacyTypeProvider : public cel::TypeProvider {
-   // created ones, the TypeInfoApis returned from this method should be the same
-   // as the ones used in value creation.
-   virtual absl::optional<const LegacyTypeInfoApis*> ProvideLegacyTypeInfo(
--      absl::string_view name) const {
-+      absl::string_view) const {
-     return absl::nullopt;
-   }
- };
-diff --git a/base/function_result.h b/base/function_result.h
-index 9bc2d67..bf6110c 100644
---- a/base/function_result.h
-+++ b/base/function_result.h
-@@ -26,7 +26,10 @@ namespace cel {
- // allows for lazy evaluation of expensive functions.
- class FunctionResult final {
-  public:
-+#pragma GCC diagnostic push
-+#pragma GCC diagnostic ignored "-Wdefaulted-function-deleted"
```

Comment:
at least last time i looked this dep desperately needs some upstream maintenance - if this has not been resovled then we really do not want to lose this

likewise if this introduces other log hogs then they should also be addressed

Reply:
The upstream is continuously updated but it uses google3 defaults which silence these warnings. The right way maybe using a different `cc_library` that allows customizing copts so that we can pass different defaults. @jcking, thoughts?

Reply:
If I get your attention, @jcking, can you triage this compiler failure on MSVC:
```
external/com_google_cel_cpp\base/internal/data.h(545): error C2607: static assertion failed
```

Reply:
re using copts, when i looked at this before i wasnt sure if it would be desirable to supress the warning entirely, given that it was one instance creating a lot of repeated logs

the problem comes back with this PR - so at least we want to put back the patch, altho i would be happy with whatever other method was used to suppress this - i think there may be some new unwanted logging issues also

Reply:
I can fix clang errors upstream. Since we're picking up the head of CEL, it'll be auto-exported soon.

Reply:
Should be silenced now.